[FM-4738] Update README to follow README template for modules

wilson208 opened the pull request:
README follows [this template](https://docs.puppet.com/puppet/latest/READMEtemplate.txt).

Also add CHANGELOG.md with high level overview of whats going into first unsupported release 0.1.0.

ferventcoder commented on CHANGELOG.md:
```diff
@@ -0,0 +1,10 @@
+  - `iis_application_pool`
+  - `iis_site`
+  - `iis_application`
+  - `iis_virtual_directory`
```
link to issues if you have them

ferventcoder commented on README.md:
```diff
@@ -1,17 +1,215 @@
-# iis
+# IIS
```
I believe @jbondpdx and group want this to be lowercase to signify this is the module, not the technology.

ferventcoder commented on README.md:
```diff
@@ -1,17 +1,215 @@
+
+### Beginning with puppetlabs-iis
+
+This module requires that IIS is already installed on your server. This could be done using the [puppet/windowsfeature module](https://forge.puppet.com/puppet/windowsfeature) by Vox Pupuli by installing the `Web-WebServer` & `Web-Scripting-Tools` Windows Features.
```
~~installing the~~ ensuring the

ferventcoder commented on README.md:
```diff
@@ -1,17 +1,215 @@
+
+#### `managedpipelinemode`
+
+The managedPipelineMode of the ApplicationPool. Must be either `Integrated` or `Classic`
```
Any specific reason "managedPipelineMode" is in camelcase?

ferventcoder commented on README.md:
```diff
@@ -1,17 +1,215 @@
+
+#### `managedruntimeversion`
+
+The managedRuntimeVersion of the ApplicationPool.
```
Same question about camel case here

ferventcoder commented on README.md:
```diff
@@ -1,17 +1,215 @@
+### Compatibility
+
+#### OS Compatibility
+This module is compatible only with `Windows Server 2008R2`, `Windows Server 2012` & `Windows Server 2012R2`. 
```
Out of the gate we should be supporting Windows Server 2016 as well. cc @jpogran @puppetlabs/modules

ferventcoder commented on README.md:
```diff
@@ -1,17 +1,215 @@
+This module is compatible only with `Windows Server 2008R2`, `Windows Server 2012` & `Windows Server 2012R2`. 
+
+#### Powershell Compatibility
+This module requires that `Powershell 3` or greater is installed on the system.
```
Since this is not the default for Windows Server 2008/R2, it should be noted as part of the setup section as well.

wilson208 commented:
@ferventcoder I have updated README and CHANGELOG with your comments. These are in separate commits rather than squashed so you can see just the changes I made based on your comments. Do a squash and merge if you are happy :)

jpogran commented on README.md:
```diff
@@ -2,16 +2,216 @@
+## Description
 
-This module adds a new provider capable of executing iis commands.
+This module provides a provider to manage IIS application pools and IIS sites.
```
Might be better to say `adds` instead of `provides`, as it sounds redundant. I would also put sites before pools, if we are going to list things, as thats order of importance.

jpogran commented on README.md:
```diff
@@ -2,16 +2,216 @@
+
+### Beginning with puppetlabs-iis
+
+This module requires that IIS is already installed on your server. This could be done using the [puppet/windowsfeature module](https://forge.puppet.com/puppet/windowsfeature) by Vox Pupuli by ensuring the `Web-WebServer` & `Web-Scripting-Tools` Windows Features.
```
`from Vox Pupuli`

`Windows Features are present`

jpogran commented on README.md:
```diff
@@ -2,16 +2,216 @@
+
+This module requires that IIS is already installed on your server. This could be done using the [puppet/windowsfeature module](https://forge.puppet.com/puppet/windowsfeature) by Vox Pupuli by ensuring the `Web-WebServer` & `Web-Scripting-Tools` Windows Features.
+
+**Note:** This module requires Powershell 3, which is not the default for Windows Server 2008R2 and will need to be installed for this module to function.
```
This is not set in stone at the moment. We haven't set PS v3 as the minimum yet.

wilson208 commented:
@jpogran I have addressed your feedback from above. I haven't edited the PS v3 minimum requirement section yet as I am unsure whether or not we will be supporting this in the initial community preview release.

PS: Ignore git commit messages, will be squashing it all once you are happy with the content.

jpogran commented on README.md:
```diff
@@ -2,16 +2,216 @@
+
+This module requires that IIS is already installed on your server. This could be done using the [puppet/windowsfeature module](https://forge.puppet.com/puppet/windowsfeature) from Vox Pupuli by ensuring the `Web-WebServer` & `Web-Scripting-Tools` Windows Features are present.
+
+**Note:** This module requires Powershell 3, which is not the default for Windows Server 2008R2 and will need to be installed for this module to function.
```
We are officially going with PowerShell v2 as minimum. Change this line to reflect this and I will merge.

wilson208 replied:
Have now updated this to specify a minimum of Powershell 2 is required.

Commit:
[FM-4738] Update README to follow README template for modules

Add CHANGELOG.md with linked JIRA tickets that are planned for release 0.1.0
